djautotask/models.py

```diff
@@ -618,6 +618,12 @@ def __str__(self):
         return self.title
 
     def update_at(self, **kwargs):
+
+        if not self.project:
+            raise api_rest.AutotaskAPIClientError(
```

Comment:
As nice as it would be to just catch this exception, `AutotaskAPIClientError` is definitely not the exception to use. As this isn't a problem with the API client. 

Is a task being in our DB a result of AT allowing tasks without projects or is that because we don't cascade delete tasks without a project? If the cause of a projectless task is somehow our fault its probably better to handle it entirely in our app rather than django-autotask.

Reply:
Yeah you're right. I was thinking just 'client' not APIClient for some reason. So client makes request to API but it can't build the URL so that's the client's fault - raise the exception. 

I think it's our fault that we allow a task without a project in our DB. Docs say project is required: https://www.autotask.net/help/DeveloperHelp/Content/APIs/REST/Entities/TasksEntity.htm . I'll test creating/updating a task via the API to see if the it enforces having a project associated with the task. If the API does have validation for this, I figure I'll just remove the `nul=True` from the `project` FK on Task and cascade delete. 

I think that should solve the issue. Let me know if are aware of reasons why this wouldn't be a good approach.

Reply:
Confirmed. You have to provide a projectId to create a task via the API or else the POST will fail. 

So I think we can remove the null=True from the project FK on Task and add cascade delete. 

That would mean we handle the problem before it becomes a problem in django-autotask by simply not saving tasks if they somehow don't have a project. If you don't see a problem with that I'll make that change.

Reply:
Due to a bug in the tracker with this version of django (known issue) we may not be able to cascade delete without getting a database error. So, be aware of that.

Reply:
Needed a data migration so added that. It just removes any tasks from the DB if they have a null project. Tested syncing a task without a project (should be impossible) but checked anyway - we'll just catch the integrity error and keep syncing. When a project is removed during a full sync I didn't see any problems with tasks not getting cascade deleted. 

Ready for another look.

Reply:
I don't think you pushed your changes